Source/PluginEditor.cpp

```diff
@@ -15,25 +15,59 @@ SubsynthAudioProcessorEditor::SubsynthAudioProcessorEditor (SubsynthAudioProcess
 {
     // Make sure that before the constructor has finished, you've set the
     // editor's size to whatever you need it to be.
-    setSize (1000, 300);
+    ///setSize (1000, 300);
     setupGain();
     
+    setSize (800, 565);
+
     waveSelect.addItem("Sine", 1);
     waveSelect.addItem("Square", 2);
     waveSelect.addItem("Saw", 3);
     waveSelect.addItem("Triangle", 4);
     waveSelect.setSelectedId(1);
-        
+            
+    filterSelect.addItem("Low Pass", 1);
+    filterSelect.addItem("Band Pass", 2);
+    filterSelect.addItem("High Pass", 3);
+    filterSelect.onChange = [this] { filterChanged(); };
+    filterSelect.setSelectedId(1);
+
+    filterCutoff.setSliderStyle(juce::Slider::SliderStyle::LinearHorizontal);
+    filterCutoff.setRange(20.0f, 20000.0f, 0.01f);
+    filterCutoff.setSkewFactorFromMidPoint(5000);
+    filterCutoff.setValue(10000.0f);
+    filterCutoff.setTextBoxStyle(juce::Slider::NoTextBox, false, 0, 0);
+    filterCutoff.setPopupDisplayEnabled(true, true, this);
+    filterCutoff.onDragEnd = [this] { filterChanged(); };
+    filterCutoff.setTextValueSuffix(" Hz");
     
+    filterRes.setSliderStyle(juce::Slider::SliderStyle::LinearHorizontal);
+    filterRes.setRange(1.0f, 5.0f, 0.1f);
+    filterRes.setValue(2.0f);
+    filterRes.setTextBoxStyle(juce::Slider::NoTextBox, false, 0, 0);
+    filterRes.setPopupDisplayEnabled(true, true, this);
+    filterRes.onDragEnd = [this] { filterChanged(); };
+
     // Expose slider to UI/Editor
     addAndMakeVisible(&waveSelect);
     addAndMakeVisible(&keyboard);
     addAndMakeVisible(&adsrSliders);
     addAndMakeVisible(&gainSlide);
     addAndMakeVisible(&gainLabel);
+    addAndMakeVisible(&filterSelect);
+    addAndMakeVisible(&filterCutoff);
+    addAndMakeVisible(&filterRes);
     
     // Add listeners
     waveSelect.addListener(this);
+    filterSelect.addListener(this);
+    filterCutoff.addMouseListener(this, true);
+    filterRes.addMouseListener(this, true);
+
+    // Waveform Visualiser
+    addAndMakeVisible(&p.wfVisualiser);
+    p.wfVisualiser.setBounds(10, 360, getWidth() - 20, 200); // add to resized() below - figure out how to access p there?
```

Comment:
`p` equivalent to the `audioProcessor`, so you can access by invoking `audioProcessor.wfVisualiser.setBounds(...)` in `resized()`